Add-Rating

WangRemy opened the pull request.

WangRemy committed:
Added song_details template

WangRemy committed:
Changed author to artist.

Its more logical to name the "author" of a song artist.

WangRemy committed:
Added some link building

When on the index page you can click on list of songs.
When on the song_list page you can click on the title
to go to the detail page of the selected song.

WangRemy committed:
added song_detail page for views

added the view for song_detail page.

WangRemy committed:
applied flake8 style

WangRemy committed:
Added Rating class and Rating Choices

Ratings vary from 1-5 / trash to excellent.
Migrated the class to the database

WangRemy committed:
Added the rating function.

Added the rating function to the details page with a form.
Checks if the request/form is a POST and if valid
the form will be saved and user is redirected to the song list page.

SanderBeekhuis commented on .gitignore:
```diff
@@ -1,3 +1,3 @@
 .idea
-*/__pycache__
+**/__pycache__
```
Nice!

SanderBeekhuis commented on .gitignore:
```diff
@@ -1,3 +1,3 @@
 .idea
-*/__pycache__
+**/__pycache__
 db.sqlite3
```
Trailing newline please :)

SanderBeekhuis commented on song_rater/views.py:
```diff
@@ -15,3 +16,24 @@ def index(request):
+RatingForm = modelform_factory(Rating, exclude=["song"])
+
+
+def song_detail(request, id):
+    if request.method == "POST":
+        form = RatingForm(request.POST)
+        if form.is_valid():
+            rating = form.save(commit=False)
+            rating.song_id = id
+            rating.save()
+            return redirect("song_rater:song_list")
+    else:
+        form = RatingForm()
+
+    context = {"details": get_object_or_404(Song, pk=id),
+               "form": form,
+               }
+
+    return render(request, 'song_rater/song_detail.html', context)
```
OPTIONAL: Have you looked if you can replace this with a GenricView by Django? Maybe `DetailView` would be a fit.  

I think it might lead to cleaner code, but I think you will touch this code again in follow-up Pull Requests. It might be best to keep this as-is and maybe try to refactor it at the end. :)

SanderBeekhuis replied:
As discussed in a call @WangRemy  won't look into this.

SanderBeekhuis reviewed (changes requested):
The code you've written looks good. Nice merge request.

It might be nice to add the Rating to the admin interface as well (but this can be part of a new MR as well) 

I'm missing tests (or an argumentation why there are none). Please write some.

SanderBeekhuis commented:
> It might be nice to add the Rating to the admin interface as well (but this can be part of a new MR as well)

It seems you did in #6  , fine by me.

WangRemy committed:
Add tests for detail view page

The test checks if the url is valid and makes a test song
with artist which the test then checks if the title is actually
displayed on the page.

WangRemy committed:
Add tests for detail view page

The test checks if the url is valid and makes a test song
with artist which the test then checks if the title is actually
displayed on the page.

SanderBeekhuis commented:
Nice test case! :)
